### DISCUSSION TITLE
Font icons not working

### DISCUSSION THREAD

**Discussion by Fr0go1** [ocornut/imgui#6907] [New Users Build/Link/Run issues ONLY!]
**_(ADMIN comment added in 2025 --> "see https://github.com/ocornut/imgui/blob/master/docs/FONTS.md")_**

----

The font icons are not working i will provide a image example using debugging

![image](https://github.com/ocornut/imgui/assets/112802612/f4ea5af1-5862-4ae5-b5aa-f7ec3d061c0e)

this is my code

#include "Utils/style.hpp"
#include "font/IconsFontAwesome4.h"

void Cheats::Menu()
{
    static int selectedStyleIndex = 0;
    static int tabb = 0;
    static int styleee = 1;

    ImGui::Begin("Menu",nullptr,ImGuiWindowFlags_AlwaysAutoResize);
    {    
        ImGuiIO& io = ImGui::GetIO();
        io.Fonts->AddFontDefault();

        ImFontConfig config;
        config.MergeMode = true;
        config.PixelSnapH = true;
        config.OversampleH = 3;
        config.OversampleV = 3;
        static const ImWchar icon_ranges[] = { ICON_MIN_FA, ICON_MAX_FA, 0 };
        io.Fonts->AddFontFromFileTTF(FONT_ICON_FILE_NAME_FA, 16.0f, &config, icon_ranges);

        ImGui::DebugTextEncoding(ICON_FA_CROSSHAIRS);
        
this is my font file
        
![image](https://github.com/ocornut/imgui/assets/112802612/4c005ac2-b13b-45fa-9650-a83acf8ac2cc)

I know that it says IconsFontAwesome4 even though the ttf is for awesome5 but the code inside the file is the correct awesome5 code

**Comment by ocornut** [comment]
You can use the Metrics->Fonts section to visualize which glyphs are loaded.

**Reply by Fr0go1** [reply]
![image](https://github.com/ocornut/imgui/assets/112802612/1f41ae65-ca02-45bf-9307-923403b4efee)
its just a bunch of 0 glyphs

**Comment by Fr0go1** [comment]
can anyone help me?

**Comment by thedmd** [comment]
You're generating new fonts every frame. Move that code to initialization, before main loop. This alone may fix the problem you have.
